### PR TITLE
Relocating external dependencies in fat jar

### PR DESCRIPTION
**Pull request by tomasbjerre**
* To avoid classpath collisions.

### CODE REVIEW

**Comment by cdancy** [comment]
This is fine but lets move to `<project-root>/gradle/additional-artifacts.gradle` where I've already stubbed out this task.

Lets do the below instead. Think that should cover all pulled in deps and give a more expected _shaded_ naming convention.

```
shadowJar {
     relocate 'com.google', 'com.cdancy.bitbucket.rest.google'
     relocate 'com.squareup', 'com.cdancy.bitbucket.rest.squareup'
     relocate 'autovalue', 'com.cdancy.bitbucket.rest.autovalue'
     relocate 'javax', 'com.cdancy.bitbucket.rest.javax'
     relocate 'net', 'com.cdancy.bitbucket.rest.net'
     relocate 'okio', 'com.cdancy.bitbucket.rest.okio'
     relocate 'org', 'com.cdancy.bitbucket.rest.org'
 }
```

**Comment by tomasbjerre** [comment]
So se.*, ch.* and bsh.* should not be relocated?

**Comment by cdancy** [comment]
I don't see either of those in the `-all` jar. Can you paste the contents of your `build/libs/*-all.jar` here? Perhaps they are different then mine. `unzip -l *-all.jar`

**Comment by tomasbjerre** [comment]
You are right =) Its not in there!

**Comment by cdancy** [comment]
Edited the code snippet above to add a `shaded` package to make it absolutely clear these packages are shaded.

File: gradle/additional-artifacts.gradle [hunk 1]
```diff
@@ -1,5 +1,15 @@
 shadowJar {
-    mergeServiceFiles()
```
Comment (cdancy) [inline gradle/additional-artifacts.gradle]: Any particular reason you removed this line or just a typo?

Comment (tomasbjerre) [inline gradle/additional-artifacts.gradle]: Added it again!

File: gradle/additional-artifacts.gradle [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+     relocate 'com.squareup', 'com.cdancy.bitbucket.rest.shaded.squareup'
+     relocate 'autovalue', 'com.cdancy.bitbucket.rest.shaded.autovalue'
+     relocate 'javax', 'com.cdancy.bitbucket.rest.shaded.javax'
+     relocate 'net', 'com.cdancy.bitbucket.rest.shaded.net'
+     relocate 'okio', 'com.cdancy.bitbucket.rest.shaded.okio'
+     relocate 'org', 'com.cdancy.bitbucket.rest.shaded.org'
+}
+
+artifacts {
+    archives shadowJar
```
Comment (tomasbjerre) [inline gradle/additional-artifacts.gradle]: Fixed

**Comment by cdancy** [comment]
I don't mind you adding in the maven/generated dirs. Can't remember exactly why I ripped that eclipse code out initially but I remember it causing problems within my workspace.

In either event: feel free to add the generated dirs to the `.gitignore` if it helps.

**Comment by tomasbjerre** [comment]
Ok I'll remember it if I need something else changed in this project! But it should be ready to be merged now right?

**Comment by cdancy** [comment]
Merged. @tomasbjerre thanks for the PR!

**Pull request merged**